### ISSUE TITLE
some of the pinned/starred data loses randomly (possibly only under wayland)

### ISSUE THREAD

**Issue by realKarthikNair** [Tudmotu/gnome-shell-extension-clipboard-indicator#322]
I'm not really sure if this is a Wayland Issue, but whenever I've noticed this issue is popping up in both Ubuntu 21.10 and Pop os beta 21.10, they were on Wayland. At this point i cant confirm whether it is reproducible while switching to X11, cuz the behavior is really random. Btw for some reasons, my work involves switching between Wayland and X11 few times a day.

**DO tell me if any logs etc is required**

System info

                 /////////////                karthik@cosmic 
             /////////////////////            -------------- 
          ///////*767////////////////         OS: Pop!_OS 21.10 x86_64 
        //////7676767676*//////////////       Host: VivoBook_ASUSLaptop X421IAY_M413IA 1.0 
       /////76767//7676767//////////////      Kernel: 5.15.5-76051505-generic 
      /////767676///*76767///////////////     Uptime: 25 mins 
     ///////767676///76767.///7676*///////    Packages: 2288 (dpkg), 4 (flatpak), 6 (snap) 
    /////////767676//76767///767676////////   Shell: bash 5.1.8 
    //////////76767676767////76767/////////   Resolution: 1920x1080 
    ///////////76767676//////7676//////////   DE: GNOME 40.5 
    ////////////,7676,///////767///////////   WM: Mutter 
    /////////////*7676///////76////////////   WM Theme: Adwaita 
    ///////////////7676////////////////////   Theme: Adwaita-dark [GTK2/3] 
     ///////////////7676///767////////////    Icons: Numix-Circle [GTK2/3] 
      //////////////////////'////////////     Terminal: x-terminal-emul 
       //////.7676767676767676767,//////      CPU: AMD Ryzen 5 4500U with Radeon Graphics (6) @ 2.375 
        /////767676767676767676767/////       GPU: AMD ATI 03:00.0 Renoir 
          ///////////////////////////         Memory: 2551MiB / 7428MiB 
             /////////////////////
                 /////////////

**Comment by Tudmotu** [comment]
Thank you for reporting this.
Are you using the latest version of this extension?

**Comment by realKarthikNair** [comment]
Yes, the latest version.

**Comment by Johann-Tree** [comment]
I have a similar issue, loosing all favorite entries randomly, on Ubuntu 18.04.6 LTS with GNOME 3.28.2 and Clipboard Indicator version 34.

**Comment by realKarthikNair** [comment]
@Johann-Tree can you confirm whether you're on Wayland or X11?

    echo $XDG_SESSION_TYPE

kindly send the output.

**Comment by Johann-Tree** [comment]
My computer is using X11.

**Comment by realKarthikNair** [comment]
Thanks for the info @Johann-Tree . So @Tudmotu it can safely be assumed that this is not a Wayland issue.

**Comment by Tudmotu** [comment]
Thanks. Interesting.

**Comment by realKarthikNair** [comment]
@Johann-Tree  can u confirm one more thing? just like me, did you also change the default history size from 15 to something bigger?

**Comment by Johann-Tree** [comment]
Yes, I changed it to 200 and some time ago I had even more, 400 or 500. But these high numbers led to instability.

**Comment by Tudmotu** [comment]
So a couple of things about the history size:

- The only reason such a large size is possible is because people insisted, but my personal opinion has always been that this is bad because the extension is not designed for large data manipulation. My recommendation is 50 items tops.
- A large enough history can slowdown and even crash the extension. This is not only the number of items, but also the size of the content.
- That's why there is also the "Max cache file size" option in the settings - it limits the byte size of the file to minimize issues

What is the "Max cache file size" you have? Maybe try increasing it will help.
To be clear - increasing it too much can cause the extension to considerably slow down.

**Comment by Tudmotu** [comment]
Let me know if this helps.
This whole limit is kinda confusing.

**Comment by realKarthikNair** [comment]
@Tudmotu thanks for the info, i'll drop down the limit to 30. also please tell me you recommended cache size on 30 count.

**Comment by Tudmotu** [comment]
I think it really depends on your machine. You can try increasing it until you start feeling slowness.
I guess up to 10,000 should be fine on modern computers. On slow computers it can be problematic, and especially the search feature will be very slow.

**Comment by realKarthikNair** [comment]
thanks

**Comment by SUPERCILEX** [comment]
Hey folks, this should be fixed in a re-written version of the extension: https://github.com/SUPERCILEX/gnome-clipboard-history.
